api/include/opentelemetry/context/runtime_context.h

```diff
@@ -175,15 +175,23 @@ class ThreadLocalContextStorage : public RuntimeContextStorage
   // Return the current context.
   Context GetCurrent() noexcept override { return GetStack().Top(); }
 
-  // Resets the context to a previous value stored in the
-  // passed in token. Returns true if successful, false otherwise
+  // Resets the context to the value previous to the passed in token. This will
+  // also detach all child contexts of the passed in token.
+  // Returns true if successful, false otherwise.
   bool Detach(Token &token) noexcept override
   {
-    if (!(token == GetStack().Top()))
+    if (!GetStack().Contains(token))
```

Comment:
I guess in 99% of the chance `GetStack().Top()` would equal to the token, so it might worth a special case optimization (might save few CPU cycles on L188?):
```c
if (token == GetStack().Top())
{
  GetStack().Pop();
  return true;
}

// check if token exists on the stack and pop items above it
```

Reply:
Good point. Added.